js/src/indexation.js

```diff
@@ -1,33 +1,123 @@
+ * @param {function(Object[], Object[])} action   The action to register.
+ *
+ * @returns {void}
+ */
+window.yoast.indexing.registerIndexingAction = ( endpoint, action ) => {
+	IndexingActions[ endpoint ] = action;
+};
+
+/**
+ * @deprecated Deprecated since 15.1. Use `window.yoast.indexing` instead.
+ */
+window.yoast.indexation = window.yoast.indexing;
+
+/**
+ * Registers a pre-indexing action on the given indexing endpoint.
+ *
+ * This action is executed before the endpoint is first called with the indexing
+ * settings as its first argument.
+ *
+ * @deprecated Deprecated since 15.1. Use `window.yoast.indexing.registerPreIndexingAction` instead.
+ *
+ * @param {string}   endpoint The endpoint on which to register the action.
+ * @param {function} action   The action to register.
+ *
+ * @returns {void}
+ */
+window.yoast.indexation.registerPreIndexationAction = ( endpoint, action ) => {
+	console.warn( "Deprecated since 15.1. Use 'window.yoast.indexing.registerPreIndexingAction' instead." );
+	window.yoast.indexing.registerPreIndexingAction( endpoint, action );
+};
+
+/**
+ * Registers an action on the given indexing endpoint.
+ *
+ * This action is executed each time after the endpoint is called, with the objects
+ * returned from the endpoint as its first argument and the indexing settings as its second argument.
+ *
+ * @deprecated Deprecated since 15.1. Use `window.yoast.indexing.registerIndexingAction` instead.
+ *
+ * @param {string}   endpoint The endpoint on which to register the action.
+ * @param {function(Object[], Object[])} action   The action to register.
+ *
+ * @returns {void}
+ */
+window.yoast.indexation.registerIndexationAction = ( endpoint, action ) => {
+	console.warn( "Deprecated since 15.1. Use 'window.yoast.indexing.registerIndexingAction' instead." );
+	window.yoast.indexing.registerIndexingAction( endpoint, action );
+};
+
+/**
+ * Indexes the site and shows a progress bar indicating the indexing process' progress.
+ */
+class Indexing extends Component {
+	/**
+	 * Indexing constructor.
+	 *
+	 * @param {Object} props The properties.
+	 */
+	constructor( props ) {
+		super( props );
+
+		this.settings = yoastIndexingData;
+		this.stoppedIndexing = false;
+
+		this.state = {
+			inProgress: false,
+			processed: 0,
+			amount: this.settings.amount,
+			error: null,
+		};
+
+		this.startIndexing = this.startIndexing.bind( this );
+		this.stopIndexing = this.stopIndexing.bind( this );
+	}
 
 	/**
-	 * Does an indexation request.
+	 * Does an indexing request.
 	 *
-	 * @param {string} url   The url of the indexation that should be done.
+	 * @param {string} url   The url of the indexing that should be done.
 	 * @param {string} nonce The WordPress nonce value for in the header.
 	 *
 	 * @returns {Promise} The request promise.
 	 */
-	async function doIndexationRequest( url, nonce ) {
+	async doIndexingRequest( url, nonce ) {
```

Comment:
Don't we need `function` here? (and in other places)

Reply:
In JavaScript classes like this, the `function` keyword is not needed for methods.
You only need it when defining a function on its own, outside of a class 🙂.